### PR TITLE
Convert Dev Guidelines to Sphinx, Add Logging Guidelines

### PR DESCRIPTION
**Pull request by iguessthislldo**

### CODE REVIEW

File: dds/DCPS/TimeDuration.h [hunk 1]
```diff
@@ -19,8 +19,8 @@ namespace DCPS {
  *
  * This wraps an ACE_Time_Value, and is designed to work with TimePoint_T.
  *
- * See the "Time" section in docs/guidelines.md for background and reasoning
- * for this class.
+ * See https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html#time
```
Comment (mitza-oci) [inline dds/DCPS/TimeDuration.h:22]: also mention `docs/internal/dev_guidelines.rst`

File: dds/DCPS/Util.h [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
- *
- *
  * Distributed under the OpenDDS License.
  * See: http://www.opendds.org/license.html
  */
 
 #ifndef OPENDDS_DCPS_UTIL_H
 #define OPENDDS_DCPS_UTIL_H
 
+#include <tao/Basic_Types.h>
```
Comment (mitza-oci) [inline dds/DCPS/Util.h:9]: Why is this needed?

File: README.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ http://download.objectcomputing.com/OpenDDS/
 http://download.opendds.org/doxygen/.
 
 - For developers wanting to contribute to OpenDDS, please take the time to read
-[the developer's guidelines](docs/guidelines.md).
+[the developer's guidelines](https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html).
```
Comment (mitza-oci) [inline README.md:51]:
```suggestion
[the development guidelines](https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html).
```
to match index.rst

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: dds/DCPS/Util.h (hunk 2 above)
Comment (iguessthislldo) [reply dds/DCPS/Util.h:9]: For this: https://github.com/objectcomputing/OpenDDS/blob/e043b21ee296e67060dabc8c69a0d43fb760e88b/dds/DCPS/Util.h#L138

Comment (mitza-oci) [reply dds/DCPS/Util.h:9]: Ok, I was looking at the `array_count` function template -- the only other part of this file that changed

**Review by mitza-oci — APPROVED** [review]
